django/contrib/syndication/views.py

```diff
@@ -82,14 +82,18 @@ def _get_dynamic_attr(self, attname, obj, default=None):
             # Check co_argcount rather than try/excepting the function and
             # catching the TypeError, because something inside the function
             # may raise the TypeError. This technique is more accurate.
-            try:
-                code = attr.__code__
-            except AttributeError:
-                code = attr.__call__.__code__
-            if code.co_argcount == 2:  # one argument is 'self'
-                return attr(obj)
-            else:
-                return attr()
+            code = attr.__code__
```

Comment:
As far as I'm aware we cannot stop catching `AttributeError`, this will crash for callable objects :thinking: , e.g.
```python
class MyCallable:
    def __call__(self):
        return 1

>>> MyCallable.__code__
AttributeError: class MyCallable has no attribute '__code__'
>>> MyCallable.__call__.__code__.co_argcount
1
```
This is currently not covered by tests, can we add a test case in the separate PR?

Reply:
Thank you for the example!

Done in https://github.com/django/django/pull/15724